solr/core/src/java/org/apache/solr/cloud/Overseer.java

```diff
@@ -219,6 +221,8 @@ else if (LeaderStatus.YES != isLeader) {
                 @Override
                 public void onEnqueue() throws Exception {
                   workQueue.offer(data);
+                  stateUpdateQueue.poll();
```

Comment:
seems as good as can readily be done given the current queue formulation -- but this kind of thing is really what ZK transactions are for! :)